recipe/patches/0004-Undo-dmlc-xgboost-9436.patch

```diff
@@ -25,31 +25,21 @@ index d6b861ab..ca0125cb 100644
 -  list(APPEND rmm_link_libs CUDA::cudart_static)
 -  set_target_properties(rmm::rmm PROPERTIES INTERFACE_LINK_LIBRARIES "${rmm_link_libs}")
 -  get_target_property(rmm_link_libs rmm::rmm INTERFACE_LINK_LIBRARIES)
- endif (PLUGIN_RMM)
+ endif()
  
- #-- library
+ if(PLUGIN_SYCL)
 diff --git a/cmake/Utils.cmake b/cmake/Utils.cmake
-index 08050205..98e96e30 100644
+index d555f5edf..8659df7c3 100644
 --- a/cmake/Utils.cmake
 +++ b/cmake/Utils.cmake
-@@ -177,8 +177,7 @@ function(xgboost_set_cuda_flags target)
-   set_target_properties(${target} PROPERTIES
-     CUDA_STANDARD 17
-     CUDA_STANDARD_REQUIRED ON
--    CUDA_SEPARABLE_COMPILATION OFF
--    CUDA_RUNTIME_LIBRARY Static)
-+    CUDA_SEPARABLE_COMPILATION OFF)
- endfunction(xgboost_set_cuda_flags)
+@@ -240,7 +240,6 @@ macro(xgboost_target_link_libraries target)
  
- macro(xgboost_link_nccl target)
-@@ -280,7 +279,6 @@ macro(xgboost_target_link_libraries target)
- 
-   if (USE_CUDA)
+   if(USE_CUDA)
      xgboost_set_cuda_flags(${target})
 -    target_link_libraries(${target} PUBLIC CUDA::cudart_static)
```

Comment:
Hyunsu, do you remember why we drop the `CUDA::cudart_static` linkage here?

Am wondering if it makes sense to keep it, but move it into `xgboost_set_cuda_flags`

Reply:
I'm not quite sure. Let's see if we can get the build working with it.